### PR TITLE
fea(internal): remove repoCache global

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

File: lib/datasource/packagist/index.ts [hunk 1]
```diff
@@ -172,19 +173,19 @@ interface AllPackages {
 }
 
 async function getAllPackages(regUrl: string): Promise<AllPackages | null> {
-  let repoCacheResult = global.repoCache[`packagist-${regUrl}`];
+  let repoCacheResult = getRepoCached(`packagist-${regUrl}`);
```
Comment (viceice) [inline lib/datasource/packagist/index.ts:176]:
```suggestion
  let repoCacheResult = getRepoCached<Promise<AllPackages | null>>(`packagist-${regUrl}`);
```

File: lib/datasource/packagist/index.ts [hunk 2]
```diff
@@ -172,19 +173,19 @@ interface AllPackages {
...
   // istanbul ignore if
   if (repoCacheResult) {
     while (repoCacheResult === 'pending') {
       await delay(200);
-      repoCacheResult = global.repoCache[`packagist-${regUrl}`];
+      repoCacheResult = getRepoCached(`packagist-${regUrl}`);
```
Comment (viceice) [inline lib/datasource/packagist/index.ts:181]:
```suggestion
      repoCacheResult = getRepoCached<Promise<AllPackages | null>>(`packagist-${regUrl}`);
```

File: lib/manager/bundler/artifacts.spec.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ describe('bundler.updateArtifacts()', () => {
     env.getChildProcessEnv.mockReturnValue(envMock.basic);
     bundlerHostRules.findAllAuthenticatable.mockReturnValue([]);
     resetPrefetchedImages();
-    global.repoCache = {};
+
```
Comment (viceice) [inline lib/manager/bundler/artifacts.spec.ts:52]:
```suggestion
```

File: lib/manager/bundler/artifacts.ts [hunk 4]
```diff
@@ -70,12 +71,12 @@ export async function updateArtifacts(
     config,
   } = updateArtifact;
   const { compatibility = {} } = config;
-
   logger.debug(`bundler.updateArtifacts(${packageFileName})`);
+  const existingError = getRepoCached('bundlerArtifactsError');
```
Comment (viceice) [inline lib/manager/bundler/artifacts.ts:75]:
```suggestion
  const existingError = getRepoCached<string>('bundlerArtifactsError');
```

File: lib/util/got/cache-get.ts [hunk 5]
```diff
@@ -26,13 +22,16 @@ export default create({
...
+        setRepoCached(
+          cacheKey,
+          next(options).catch((err) => {
+            setRepoCached(cacheKey, null);
+            throw err;
+          })
+        );
       }
-      return global.repoCache[cacheKey].then((response) => ({
+      return getRepoCached(cacheKey).then((response) => ({
```
Comment (viceice) [inline lib/util/got/cache-get.ts:34]:
```suggestion
      return getRepoCached<Promise<any>>(cacheKey).then((response) => ({
```

**Review by viceice — CHANGES_REQUESTED** [review]
Some small suggestions. Finally another global removed

File: lib/datasource/packagist/index.ts [hunk 6]
```diff
@@ -172,19 +172,8 @@ interface AllPackages {
 }
 
 async function getAllPackages(regUrl: string): Promise<AllPackages | null> {
-  let repoCacheResult = global.repoCache[`packagist-${regUrl}`];
-  // istanbul ignore if
-  if (repoCacheResult) {
-    while (repoCacheResult === 'pending') {
```
Comment (rarkins) [inline lib/datasource/packagist/index.ts:178]: This was really ugly..

**Pull request merged**